File: app/src/main/java/com/webviewblock/app/SharedPreferenceProvider.java
package com.webviewblock.app;

public interface SharedPreferenceProvider {
    String getString(String key, String defValue);

    void putString(String key, String value);

    boolean getBoolean(String key, Boolean defValue);

    void putBoolean(String key, Boolean value);

    int getInt(String key, Integer defValue);

    void putInt(String key, Integer value);

    void putLong(String key, Long value);

    long getLong(String key, Long defValue);

    void clearAll();
}

File: app/src/main/java/com/webviewblock/app/SharedPreferenceProvider.kt
package com.webviewblock.app

interface SharedPreferenceProvider {
    fun getString(key: String, defValue: String): String
    fun putString(key: String, value: String)
    fun getBoolean(key: String, defValue: Boolean): Boolean
    fun putBoolean(key: String, value: Boolean)
    fun getInt(key: String, defValue: Int): Int
    fun putInt(key: String, value: Int)
    fun putLong(key: String, value: Long)
    fun getLong(key: String, defValue: Long): Long
    fun clearAll()
}